### PR TITLE
Add beta banner to recruit translators

### PR DESCRIPTION
**Pull request by nielsmaerten**
## Description
This is my first attempt at adding a beta banner to the settings page.  

## Motivation and Context
I hope this will help attracting more translators to help us finalize the (Spanish) translation initially, and other translations down the line.

## How Has This Been Tested?
Simple snapshot test. Maybe needs a bit more extensive testing?

## Additional context
image

## Types of changes

* New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I've made my changes in a separate branch
- [ ] My change is passing new and existing tests

### CODE REVIEW

File: gluco-check-frontend/src/App.tsx [hunk 1]
```diff
@@ -147,6 +148,7 @@ export default function App() {
               {user && (
                 <Route path={`${localesParamString}/settings`}>
                   <FirebaseUserDocumentContext.Provider value={docPath}>
+                    <Banner />
```
Comment (nielsmaerten) [inline gluco-check-frontend/src/App.tsx:151]: TODO: Only show this banner when the active language is in Beta (add to enum)

File: gluco-check-frontend/src/components/Banner.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  const theme = useTheme();
+  const alertVariant = theme.palette.type === "light" ? "outlined" : "filled";
+  const classes = useStyles();
+  const { t, i18n } = useTranslation();
+
+  return (
+    <Alert className={classes.alert} variant={alertVariant} severity="warning">
+      <AlertTitle>{t("settings.betaBanner.title")}</AlertTitle>
+      {
+        // TODO: Properly interpolate this so the link is clickable and takes
```
Comment (nielsmaerten) [inline gluco-check-frontend/src/components/Banner.tsx:32]: TODO

**Comment by nielsmaerten** [comment]
Hey @ddamico ,

I took a crack at adding a banner on top of the settings form. The banner appears for languages that are currently being tested (so only Spanish for now) and tries to get bilingual users to sign up to the beta.
If this works we could recruit proofreaders for new languages and speed up their release. Maybe in the future, we could even machine-translate all target languages and then have beta testers clean up the strings. That could possibly lower the "barrier-to-translations".

Can you take a look at this and tell me your thoughts? I'm still trying to get the hang of React, so apologies if this isn't up to standards ;)

Niels

**Comment by ddamico** [comment]
Hey @nielsmaerten! Was looking through the MUI docs and found that there does exist an official banner treatment, it's just not supported in the react component library we're using. Was able to put something together using the base components that _are_ part of the library and have a couple of options...

Screen Shot 2021-07-26 at 6 34 24 PM

Screen Shot 2021-07-26 at 6 38 14 PM

Definitely need better font weight on the action button, but overall what do you think? Happy to polish this up and commit it if you are into it.

**Comment by ddamico** [comment]
I think I have flip-flopped overnight about the different banner styling, let me know if you prefer what I suggested @nielsmaerten but if not I think I prefer your treatment, and I'd be happy to finish up the tests in this branch.

**Comment by nielsmaerten** [comment]
Hey @ddamico! I thought this new approach looked pretty good :) The alert component I'm using is part of MUI's 'Lab collection' meaning it's not fully stable yet. Styling in particular seems a bit off. Can you commit yours to a different branch so I can see the code? I'm also curious to see what it looks like in dark mode.

**Comment by nielsmaerten** [comment]
@ddamico One remark on your approach is that `betaBanner.title` seems to be gone? We should probably keep that, or at least the message "This translation is not available yet" to avoid confusion. Signing up for the beta is the only way a user can get access to an unfinished language

**Comment by ddamico** [comment]
> @ddamico One remark on your approach is that `betaBanner.title` seems to be gone? We should probably keep that, or at least the message "This translation is not available yet" to avoid confusion. Signing up for the beta is the only way a user can get access to an unfinished language

Yup thanks for catching that, missed the title in my initial test but I have put it back.

I've pushed a new branch, `feature/beta-banner-v2` (base branch is `feature/beta-banner`) containing my updates, have a look and let me know what you think.

**Comment by nielsmaerten** [comment]
Hey @ddamico 
I ended up going with your full width design:

**Comment by nielsmaerten** [comment]
Hey @ddamico 
I ended up going with your full-width design. Just changed the accent color to orange to make it stand out a bit more, and wrapped the icon in a circle to more closely match Material guidelines.
image

File: gluco-check-frontend/src/components/Banner.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    "&:hover": {
+      backgroundColor: theme.palette.warning.dark,
+    },
+  },
+}));
+
+function Banner() {
+  const { t, i18n } = useTranslation();
+
+  // TODO: can't we just get this straight from i18n?
```
Comment (nielsmaerten) [inline gluco-check-frontend/src/components/Banner.tsx:44]: I don't think we can. i18next has no concept of the languages it's translating to. It just takes in a language code like 'en' uses it to substitute placeholders with the strings in the corresponding json file

**Pull request merged**